st2common/st2common/hooks.py

```diff
@@ -1,376 +0,0 @@
-        else:
-            origin_allowed = list(origins)[0]
-
-        methods_allowed = ['GET', 'POST', 'PUT', 'DELETE', 'OPTIONS']
-        request_headers_allowed = ['Content-Type', 'Authorization', 'X-Auth-Token',
-                                   HEADER_API_KEY_ATTRIBUTE_NAME, REQUEST_ID_HEADER]
-        response_headers_allowed = ['Content-Type', 'X-Limit', 'X-Total-Count',
-                                    REQUEST_ID_HEADER]
-
-        headers['Access-Control-Allow-Origin'] = origin_allowed
-        headers['Access-Control-Allow-Methods'] = ','.join(methods_allowed)
-        headers['Access-Control-Allow-Headers'] = ','.join(request_headers_allowed)
-        headers['Access-Control-Expose-Headers'] = ','.join(response_headers_allowed)
-        if not headers.get('Content-Length') \
-                and not headers.get('Content-type', '').startswith('text/event-stream'):
-            headers['Content-Length'] = str(len(state.response.body))
-
-    def on_error(self, state, e):
-        if state.request.method == 'OPTIONS':
-            return webob.Response()
-
-
-class AuthHook(PecanHook):
-
-    def before(self, state):
-        # OPTIONS requests doesn't need to be authenticated
-        if state.request.method == 'OPTIONS':
-            return
-
-        # Token request is authenticated separately.
-        if (state.request.method == 'POST' and
-                re.search(AUTH_TOKENS_URL_REGEX, state.request.path)):
-            return
-
-        if cfg.CONF.auth.enable:
-            user_db = self._validate_creds_and_get_user(request=state.request)
-
-            # Store related user object in the context. The token is not passed
-            # along any longer as that should only be used in the auth domain.
-            state.request.context['auth'] = {
-                'user': user_db
-            }
-
-        if QUERY_PARAM_ATTRIBUTE_NAME in state.arguments.keywords:
-            del state.arguments.keywords[QUERY_PARAM_ATTRIBUTE_NAME]
-
-        if QUERY_PARAM_API_KEY_ATTRIBUTE_NAME in state.arguments.keywords:
-            del state.arguments.keywords[QUERY_PARAM_API_KEY_ATTRIBUTE_NAME]
-
-    def on_error(self, state, e):
-        if isinstance(e, (auth_exceptions.NoAuthSourceProvidedError,
-                          auth_exceptions.MultipleAuthSourcesError)):
-            LOG.error(str(e))
-            return self._abort_unauthorized(str(e))
-        if isinstance(e, auth_exceptions.TokenNotProvidedError):
-            LOG.exception('Token is not provided.')
-            return self._abort_unauthorized(str(e))
-        if isinstance(e, auth_exceptions.TokenNotFoundError):
-            LOG.exception('Token is not found.')
-            return self._abort_unauthorized(str(e))
-        if isinstance(e, auth_exceptions.TokenExpiredError):
-            LOG.exception('Token has expired.')
-            return self._abort_unauthorized(str(e))
-        if isinstance(e, auth_exceptions.ApiKeyNotProvidedError):
```

Comment:
What happened to this code, where do we handle this right now?

I've seen some code in `abort_request`, but I guess I'm still missing where exception to friendly user message translation happens.